src/main/java/analysis/emissions/LinkEmissionsHandler.java

```diff
@@ -0,0 +1,48 @@
+package analysis.emissions;
+
+import org.matsim.api.core.v01.Id;
+import org.matsim.api.core.v01.network.Link;
+import org.matsim.contrib.emissions.Pollutant;
+import org.matsim.contrib.emissions.events.ColdEmissionEvent;
+import org.matsim.contrib.emissions.events.ColdEmissionEventHandler;
+import org.matsim.contrib.emissions.events.WarmEmissionEvent;
+import org.matsim.contrib.emissions.events.WarmEmissionEventHandler;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class LinkEmissionsHandler implements WarmEmissionEventHandler, ColdEmissionEventHandler {
+
+    private final Map<Id<Link>, Map<Pollutant, Double>> link2pollutants = new HashMap<>();
```

Comment:
For me this class lloks (more or less?) like a copy of https://github.com/matsim-scenarios/matsim-berlin/blob/5.5.x_airPollutionComputation/src/main/java/org/matsim/analysis/EmissionsOnLinkHandler.java 

Before we spread it (more) all over the place, I suggest to either put it into the emission contrib or to the vsp-contrib. 

Since it  is somehow useful for everyone, the emissions contrib seems to be the better choice ;)

(and then maybe go through our different scenarios and remove it from there..)

If not moving it to a central place: Try to reduce private footprint and make it package-protected if possible

Reply:
For me putting it into the emissions contrib seems like a good idea, too. If it already exists even better

Reply:
I will take a look at this. It makes sense to me, too.

Reply:
Great! If you try to push directly to the main branch of matsim libs you will quickly notice that it's not possible. The only way to push is to create a new branch with your changes and then create a PR for this branch with some description.